submissions/iva-stasia/frogger-game/app.js

```diff
@@ -0,0 +1,184 @@
+"use strict";
+
+const INIT_VAL = {
+    playerX: 202,
+    playerY: 410,
+    stepX: 101,
+    stepY: 85,
+    enemyX: -101,
+    enemyY: [63, 147, 230],
+    enemyMinSpeed: 100,
+    enemyMaxspeed: 200,
+    collision: 75,
+    canvasWidth: 505,
+    canvasHeight: 606,
+};
+
+const allPlayerSprites = [
+    'images/char-boy.png',
+    'images/char-cat-girl.png',
+    'images/char-princess-girl.png'
+];
+
+let levelSpeed = 0;
+let level = 0;
+
+/*---------- DOM elements ----------*/
+
+const body = document.body;
+
+body.insertAdjacentHTML('afterbegin',
+    `
+        <div class='level'>Level: ${level}/5</div>
+        <div class='win_message'><h2>You won :)</h2><span>press enter</span></div>
+        <div class='lost_message'><h2>You lost :(</h2><span>press enter</span></div>
+        <div class='player_option'></div>
+    `);
+    
+const levelDiv = document.querySelector('.level');
+const winMessage = document.querySelector('.win_message');
+const lostMessage = document.querySelector('.lost_message');
+const playerOptionDiv = document.querySelector('.player_option');
+
+playerOptionDiv.innerHTML =
+    `
+        <h1>Choose your character</h1>
+        <img src='images/char-boy.png' class='player_img' data-player-num = '0' />
+        <img src='images/char-cat-girl.png' class='player_img' data-player-num = '1' />
+        <img src='images/char-princess-girl.png' class='player_img' data-player-num = '2' />
+    `;
+
+/*---------- Enemy ----------*/
+
+const Enemy = function(x, y, speed) {
+    this.x = x;
+    this.y = y;
+    this.speed = speed;
+    this.sprite = 'images/enemy-bug.png';
+};
+
+Enemy.prototype.update = function(dt) {
+    this.x += this.speed * dt + levelSpeed;
+
+    if (this.x > INIT_VAL.canvasWidth
+        || !playerOptionDiv.classList.contains('hide')
+        || lostMessage.classList.contains('active_message')
+        || winMessage.classList.contains('active_message')) {
+        this.x = INIT_VAL.enemyX;
+    };
+
+    this.checkCollision();
+};
+
+Enemy.prototype.checkCollision = function() {
+    if (player.x < this.x + INIT_VAL.collision &&
+        player.x + INIT_VAL.collision > this.x &&
+        player.y < this.y + INIT_VAL.collision &&
+        INIT_VAL.collision + player.y > this.y) {
+            player.resetGame();
+    };
+};
+
+Enemy.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+const allEnemies = [];
+
+INIT_VAL.enemyY.forEach(y => {
+    let enemySpeed = Math.round(Math.random() * INIT_VAL.enemyMaxspeed + INIT_VAL.enemyMinSpeed);
+    let enemy = new Enemy(INIT_VAL.enemyX, y, enemySpeed);
```

Comment:
Why `let`?

Reply:
When I wrote this code, I thought that inside the methods that execute a function for each element of the array, you need to declare variables using <code>let</code>. Now I already know that this is not so. I'll fix it.

Reply:
:white_check_mark: